atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2596,9 +2604,29 @@ public TransactionReadInfo getReadInfo() {
         return keyValueService.getOverallReadInfo();
     }
 
-    // TODO(aalouane): implement and call this on transaction exit
     @Override
-    public void reportExpectationsCollectedData() {}
+    public void reportExpectationsCollectedData() {
+        reportExpectationsCollectedData(this, expectationsDataCollectionMetrics, isStillRunning());
+    }
+
+    @VisibleForTesting
+    static void reportExpectationsCollectedData(
+            ExpectationsAwareTransaction transaction, ExpectationsMetrics metrics, boolean isStillRunning) {
```

Comment:
discussed offline: boolean param is probably unnecessary, I think it's ok to not have coverage for `log.error(...); return;`